models/CalendarEntry.php

```diff
@@ -664,4 +664,12 @@ public function getBadge()
 
         return null;
     }
+
+    public function downloadIcs()
+    {
+        $module = Yii::$app;
+        $timezone = $module->settings->get('timeZone');
+        $ics = new \humhub\modules\calendar\models\ICS($this->title, $this->description,$this->start_datetime, $this->end_datetime, null, null, $timezone);
+        return Yii::$app->response->sendContentAsFile($ics, uniqid() . '.ics', ['mimeType' => 'text/calendar']);
```

Comment:
Could you please rename this method to createICS or generateICS and remove the sendContentAsFile call? Build the ics from calendar entry is something that belongs to models/CalendarEntry but the logic for offering a file to the user should be in the controller. The return of the method should be an object of type ICS.